### PR TITLE
[Test][Doc] Add numerical tests and operator doc for the MRV2 resample kernels

### PR DESCRIPTION
**Pull request by MrMC-Sqz**
## Why

`_resample_kernel` and `_npu_gumbel_block_argmax` (`vllm_ascend/worker/v2/spec_decode/rejection_sampler_utils.py`) have no accuracy test and no operator doc.

The existing `tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_rejection_sample.py` is the model_runner **v1** path (`vllm_ascend/ops/triton/reject_sample.py`), not these kernels — the similar file name makes this easy to miss. No test imports `vllm_ascend.worker.v2.spec_decode` today.

## What

- **Operator doc**: `vllm_ascend/ops/triton/docs/resample.md`, structured as Description / Parameters / Constraints / Origin and Differences / Test Cases.
- **Accuracy test**: `tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py` — 10 cases against a PyTorch fp32 reference, covering the three residual branches (bonus, draft-logits residual, one-hot exclusion), the greedy early-return path, and one end-to-end pass through `rejection_sample`.

`_npu_gumbel_block_argmax` is a `@triton.jit` device function inlined into `_resample_kernel`, not a separately dispatched operator, so it is covered through `_resample_kernel` rather than tested on its own. The test file defines no Triton kernel and reproduces no part of the operator: cases that need an exact expected token are constructed so each vocabulary block holds a single finite residual, which the Gumbel noise cannot displace, and cases that need the residual values compare sampling frequencies over 16,384 draws against the analytic distribution.

## Notes

- Verified on Atlas A2, Python 3.11, torch_npu 2.10 — all 10 cases pass:

```bash
pytest -sv tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py
```

- Documentation and tests only; no operator behaviour is changed.
- Constraints worth flagging from the doc: `draft_sampled` is read at `resample_token_idx + 1` (the input-id stream sits one slot ahead of the logits); the `temp == 0 and not is_bonus` early return is a contract with the upstream `_insert_resampled_kernel`, which skips on the identical condition and reuses the target argmax already written.

- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

**Review by SOMEONEUNSEEN — APPROVED** [review]

**Review by herizhen — APPROVED** [review]

**Review by freyfwt — APPROVED** [review]

File: tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py [hunk 1]
```diff
@@ -0,0 +1,1136 @@
...
+
+
+# ---------------------------------------------------------------------------
+# Probe kernels (test-only harness)
+# ---------------------------------------------------------------------------
+
+
+@triton.jit
+def _gumbel_probe_kernel(
+    out_value_ptr,
```
Comment (AuroraEmiya) [inline tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py]: why define a triton kernel here in a test file ?

File: tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py
Comment (AuroraEmiya) [inline tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py:1]: As for ut , you might have to import those implementations with their source paths no by copying the actual codes.

**Review by AuroraEmiya — CHANGES_REQUESTED** [review]

File: tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py (hunk 1 above)
Comment (MrMC-Sqz) [reply tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py]: ’ _npu_gumbel_block_argmax ‘ is a triton.jit device function and cannot be launched directly from Python.
' _npu_gumbel_block_argmax ' imported from rejection_sampler_utils.py

File: tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py
Comment (MrMC-Sqz) [reply tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py:1]: The implementations under test are imported directly from the source. The test-only Triton kernel is just a launch wrapper because `_npu_gumbel_block_argmax` cannot be launched directly from Python; it does not copy its implementation.

File: tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py (hunk 1 above)
Comment (MrMC-Sqz) [reply tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py]: Removed the test-only Triton kernels. `_npu_gumbel_block_argmax` is now covered through the production `_resample_kernel` path.

File: tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py
Comment (MrMC-Sqz) [reply tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py:1]: Removed the remaining test-only Triton kernel and duplicated RNG logic. The tests now import and exercise the production `_resample_kernel` directly, using deterministic, invariant, and statistical checks.

**Review by AuroraEmiya — APPROVED** [review]
LGTM.

**Review by liqian79 — APPROVED** [review]

**Review by ZT-AIA — APPROVED** [review]
